File: 0-5_Development_ProjectGrid/action/PROJECT_GRID_REVISED/grid/setup_database.py
#!/usr/bin/env python3
"""
Setup PROJECT_GRID_REVISED Database in Supabase
Executes SQL files in order using psycopg2
"""

import os
import sys

# Supabase PostgreSQL connection string (Direct connection)
SUPABASE_CONNECTION = "postgresql://postgres:jCWzY7SKmbAV5ESZ@db.ooddlafwdpzgxfefgsrx.supabase.co:5432/postgres"

# SQL files to execute in order
SQL_FILES = [
    'project_grid_revised_36_schema.sql',
    'phase_gates_schema.sql',
    'project_grid_revised_36_data.sql',
    'phase_gates_data.sql'
]

def execute_sql_file(cursor, filename):
    """Execute a SQL file"""
    print(f"\n[*] Executing {filename}...")

    try:
        filepath = os.path.join(os.path.dirname(__file__), filename)
        with open(filepath, 'r', encoding='utf-8') as f:
            sql = f.read()

        # Execute the SQL
        cursor.execute(sql)

        print(f"[OK] {filename} executed successfully")
        return True
    except Exception as e:
        print(f"[ERROR] Error executing {filename}: {e}")
        return False

def main():
    """Main function"""
    print("[START] PROJECT_GRID_REVISED Database Setup...\n")
    print(f"[DIR] Working directory: {os.path.dirname(__file__)}\n")

    try:
        import psycopg2
    except ImportError:
        print("[INSTALL] psycopg2 not installed. Installing...")
        os.system(f"{sys.executable} -m pip install psycopg2-binary")
        import psycopg2

    try:
        # Connect to Supabase PostgreSQL
        print("[CONNECT] Connecting to Supabase...")
        conn = psycopg2.connect(SUPABASE_CONNECTION)
        conn.autocommit = False
        cursor = conn.cursor()
        print("[OK] Connected to Supabase\n")

        success_count = 0
        fail_count = 0

        # Execute each SQL file
        for filename in SQL_FILES:
            if execute_sql_file(cursor, filename):
                conn.commit()
                success_count += 1
            else:
                conn.rollback()
                fail_count += 1

        # Close connection
        cursor.close()
        conn.close()

        # Print summary
        print('\n' + '=' * 50)
        print(f"[SUCCESS] Successful: {success_count}")
        print(f"[FAILED] Failed: {fail_count}")
        print('=' * 50)

        if fail_count == 0:
            print('\n[COMPLETE] DATABASE SETUP COMPLETE!')
            print('\nNext steps:')
            print('1. Verify tables in Supabase Dashboard')
            print('2. Check project_grid_tasks_revised table has 36 rows')
            print('3. Check phase_gates table has 6 rows')
            return 0
        else:
            print('\n[WARNING] Some SQL files failed to execute.')
            return 1

    except Exception as e:
        print(f"\n[FATAL ERROR] Fatal error: {e}")
        return 1

if __name__ == '__main__':
    sys.exit(main())

File: 0-5_Development_ProjectGrid/action/PROJECT_GRID_REVISED/grid/setup-database.js
// Setup Supabase Database for PROJECT_GRID_REVISED
// Executes all SQL files in order

const { createClient } = require('@supabase/supabase-js');
const fs = require('fs');
const path = require('path');

// Supabase credentials
const SUPABASE_URL = 'https://ooddlafwdpzgxfefgsrx.supabase.co';
const SUPABASE_SERVICE_KEY = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6Im9vZGRsYWZ3ZHB6Z3hmZWZnc3J4Iiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc2MDU5MjQzNCwiZXhwIjoyMDc2MTY4NDM0fQ.qiVzF8VLQ9jyDvv5ZLdw_6XTog8aAUPyJLkeffsA1qU';

// Create Supabase client
const supabase = createClient(SUPABASE_URL, SUPABASE_SERVICE_KEY);

// SQL files to execute in order
const SQL_FILES = [
  'project_grid_revised_36_schema.sql',
  'phase_gates_schema.sql',
  'project_grid_revised_36_data.sql',
  'phase_gates_data.sql'
];

async function executeSQLFile(filename) {
  console.log(`\n📄 Executing ${filename}...`);

  try {
    const filePath = path.join(__dirname, filename);
    const sql = fs.readFileSync(filePath, 'utf8');

    // Split SQL into individual statements (simple split by semicolon)
    const statements = sql
      .split(';')
      .map(s => s.trim())
      .filter(s => s.length > 0 && !s.startsWith('--'));

    console.log(`   Found ${statements.length} SQL statements`);

    for (let i = 0; i < statements.length; i++) {
      const statement = statements[i];
      console.log(`   Executing statement ${i + 1}/${statements.length}...`);

      // Use rpc call to execute raw SQL
      const { data, error } = await supabase.rpc('exec_sql', { sql_query: statement });

      if (error) {
        // Try direct query if rpc fails
        const { error: queryError } = await supabase.from('_sql').select().throwOnError();
        if (queryError) {
          console.error(`   ❌ Error:`, error.message);
          throw error;
        }
      }
    }

    console.log(`✅ ${filename} executed successfully`);
    return true;
  } catch (error) {
    console.error(`❌ Failed to execute ${filename}:`, error.message);
    return false;
  }
}

async function setupDatabase() {
  console.log('🚀 Starting PROJECT_GRID_REVISED Database Setup...\n');
  console.log(`📍 Supabase URL: ${SUPABASE_URL}`);
  console.log(`📁 Working directory: ${__dirname}\n`);

  let successCount = 0;
  let failCount = 0;

  for (const filename of SQL_FILES) {
    const success = await executeSQLFile(filename);
    if (success) {
      successCount++;
    } else {
      failCount++;
    }
  }

  console.log('\n' + '='.repeat(50));
  console.log(`✅ Successful: ${successCount}`);
  console.log(`❌ Failed: ${failCount}`);
  console.log('='.repeat(50));

  if (failCount === 0) {
    console.log('\n🎉 DATABASE SETUP COMPLETE!');
    console.log('\nNext steps:');
    console.log('1. Verify tables in Supabase Dashboard');
    console.log('2. Check project_grid_tasks_revised table has 36 rows');
    console.log('3. Check phase_gates table has 6 rows');
  } else {
    console.log('\n⚠️  Some SQL files failed to execute.');
    console.log('Please check the errors above and try again.');
  }
}

// Run setup
setupDatabase().catch(error => {
  console.error('Fatal error:', error);
  process.exit(1);
});
